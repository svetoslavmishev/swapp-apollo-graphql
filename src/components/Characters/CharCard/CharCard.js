import React, { useContext } from "react";
import PropTypes from "prop-types";
import { Paper, Typography, ButtonBase } from "@material-ui/core";

import { ThemeContext } from "../../../themeContext";
import styles from "./CharCardStyles";

const CharCard = ({ chars: { id, image, name }, onClick }) => {
  const { currentTheme } = useContext(ThemeContext);
  const classes = styles({ currentTheme });

  return (
    <Paper className={`flex  ${classes.paper}`} onClick={onClick}>
      <div>
        <ButtonBase className={classes.image}>
          <img className={classes.img} alt="hero" src={image} />
        </ButtonBase>
      </div>
      <div className="flex w-full items-center pl-4">
        <Typography className={classes.charTitle}>{name}</Typography>
      </div>
    </Paper>
  );
};

CharCard.propTypes = {
  chars: PropTypes.shapeOf({
    id: PropTypes.string,
    image: PropTypes.string,
    name: PropTypes.string
  }),
  onClick: PropTypes.func
};

export default CharCard;
