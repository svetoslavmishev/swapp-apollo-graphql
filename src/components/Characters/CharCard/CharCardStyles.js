import { makeStyles } from '@material-ui/styles';

const styles = makeStyles(theme => ({
  paper: {
    width: 400,
    height: 90,
    margin: '.2rem',
    background: ({ currentTheme }) =>
      theme.palette[currentTheme].Cards.backgroundColor,
    border: ({ currentTheme }) =>
      `1px solid ${theme.palette[currentTheme].Cards.borderColor}`,
    '&:hover': {
      cursor: 'pointer'
    }
  },
  image: {
    width: 90,
    height: 90
  },
  charTitle: {
    fontFamily: 'StarWars',
    color: ({ currentTheme }) =>
      theme.palette[currentTheme].PrimaryHeading.fontColor
  }
}));

export default styles;
