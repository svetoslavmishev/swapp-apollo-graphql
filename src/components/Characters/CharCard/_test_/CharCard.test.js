import React from 'react';
import { shallow } from 'enzyme';
import { ThemeContext, theme } from '../../../../themeContext';
import CharCard from '../CharCard';

describe('<CharCard />', () => {
  it('renders <CharCard /> without error', () => {
    const wrapper = shallow(
      <ThemeContext.Provider value={theme}>
        <CharCard />
      </ThemeContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
