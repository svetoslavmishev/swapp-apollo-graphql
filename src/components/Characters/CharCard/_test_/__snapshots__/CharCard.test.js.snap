// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharCard /> renders <CharCard /> without error 1`] = `ShallowWrapper {}`;
